### ISSUE TITLE
Adapt Arena for Dalvik

### ISSUE THREAD

**Issue by h4ng3r** [radare/radare2#6838]
Dalvik is register-based and each method uses a different number of registers, so we need to adapt the arena in order to support this behaviour for both anal and ESIL.

More info from Google/Android [:](https://source.android.com/devices/tech/dalvik/dalvik-bytecode.html)
```The machine is register-based, and frames are fixed in size upon creation. Each frame consists of a particular number of registers (specified by the method) as well as any adjunct data needed to execute the method, such as (but not limited to) the program counter and a reference to the .dex file that contains the method.```

**Comment by radare** [comment]
From what we understood in https://hackmd.io/EYQwpgDALGbAtBArBAHPKqDsBmewA2ARjyIBMAzECg7TJAoA?edit 

we should just have an arena of a specific size.. and implement `...` in r_reg to allow resizing if needed, by expanding the last register following the same name pattern.

i did a quick test with dexdump and i found some that some dex have like 659 registers in a single function xDDDD so yep we need to implement `...` in r_reg :) to make arenas resizables

**Comment by radare** [comment]
The thing is... when entering/quiting function we shouldnt shrink down the arena, we should just shrink what the user sees in the arena.. this is .. faking down the arena size, but not removing the data in the original arena. but in fact in dalvik unused registers will not be reusable between function executions, because every time you enter a function you start with a clean state, right?

**Comment by radare** [comment]
we need toimplement the dalvik debugger in r2 to understand this :P

**Comment by h4ng3r** [comment]
@radare Yep you start with a clean state. Each function define the amount of registers needed using .register or .locals. This includes the arguments (1 or 2 register per argument) and this if the invoke is not static. This information is being parsed by Dex RBIn we just need to save it, but I dont know which is the right place.

**Comment by radare** [comment]
i guess that its starting with cleanedup register values.. or it is inherting some values from previoues calls, etc?
maybe we should add a new esil primitive to do this without having to push N regs by hand in every esil expression etc

**Comment by alvarofe** [comment]
600 registers omg and @h4ng3r said in private the arch support until 65535. So yes, defining `...` in r_reg sounds the way to fix this

**Comment by radare** [comment]
DUUUUUUUUDES

**Comment by h4ng3r** [comment]
@radare I'll try to have a first version for this release.
